app/src/main/java/org/simple/clinic/teleconsultlog/medicinefrequency/MedicineFrequencyBottomSheet.kt

```diff
@@ -0,0 +1,131 @@
+import android.os.Bundle
+import android.util.Log
+import com.jakewharton.rxbinding3.view.clicks
+import io.reactivex.Observable
+import io.reactivex.rxkotlin.ofType
+import kotlinx.android.synthetic.main.sheet_medicine_frequency.*
+import org.simple.clinic.ClinicApp
+import org.simple.clinic.R
+import org.simple.clinic.mobius.MobiusDelegate
+import org.simple.clinic.teleconsultlog.drugduration.DrugDuration
+import org.simple.clinic.teleconsultlog.medicinefrequency.di.MedicineFrequencyComponent
+import org.simple.clinic.util.unsafeLazy
+import org.simple.clinic.widgets.BottomSheetActivity
+import javax.inject.Inject
+
+class MedicineFrequencyBottomSheet : BottomSheetActivity(), MedicineFrequencySheetUiActions {
+
+  @Inject
+  lateinit var effectHandlerFactory: MedicineFrequencyEffectHandler.Factory
+
+  private lateinit var component: MedicineFrequencyComponent
+
+  companion object {
+    private const val DRUG_DURATION = "drugDuration"
+    private const val MEDICINE_FREQUENCY = "medicineFrequency"
+    private const val SAVED_MEDICINE_FREQUENCY = "savedmedicineFrequency"
+
+    fun intent(
+        context: Context,
+        drugDuration: DrugDuration,
+        medicineFrequency: MedicineFrequencies
+    ): Intent {
+      return Intent(context, MedicineFrequencyBottomSheet::class.java).apply {
+        putExtra(DRUG_DURATION, drugDuration)
+        putExtra(MEDICINE_FREQUENCY, medicineFrequency)
+      }
+    }
+  }
+
+  private val drugDuration by unsafeLazy {
+    intent.getParcelableExtra<DrugDuration>(DRUG_DURATION)!!
+  }
+
+  private val medicineFrequency by unsafeLazy {
+    intent.getSerializableExtra(MEDICINE_FREQUENCY)
+  }
+
+  private val events by unsafeLazy {
+    saveClicks()
+  }
+
+  private val delegate by unsafeLazy {
+    MobiusDelegate.forActivity(
+        events = events.ofType(),
+        defaultModel = MedicineFrequencyModel.create(MedicineFrequencies.BD),
+        init = MedicineFrequencyInit(),
+        update = MedicineFrequencyUpdate(),
+        effectHandler = effectHandlerFactory.create(this).build()
+    )
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.sheet_medicine_frequency)
+    medicineFrequencyTitleTextView.text = getString(R.string.drug_duration_title, drugDuration.name, drugDuration.dosage)
+  }
+
+  private fun saveClicks(): Observable<MedicineFrequencyEvent> {
+    return saveMedicineFrequencyButton
+        .clicks()
+        .map {
+          when {
+            medicineFrequencyOdRadioButton.isChecked -> medicineFrequency == MedicineFrequencies.OD
+            medicineFrequencyBdRadioButton.isChecked -> medicineFrequency == MedicineFrequencies.BD
+            medicineFrequencyTdsRadioButton.isChecked -> medicineFrequency == MedicineFrequencies.TDS
+            else -> medicineFrequency == MedicineFrequencies.QDS
+          }
+          SaveMedicineFrequencyClicked(medicineFrequency as MedicineFrequencies)
```

Comment:
`medicineFrequency` here is not the latest value that is selected. You are just passing the value that is passed in as the screen extra.

One way to do this is, we can define a map of radio button id to `MedicineFrequency`
```kt
  private val radioIdToMedicineFrequency = mapOf(
      R.id.medicineFrequencyOdRadioButton to OD,
      R.id.medicineFrequencyBdRadioButton to BD,
      R.id.medicineFrequencyTdsRadioButton to TDS,
      R.id.medicineFrequencyQdsRadioButton to QDS
  )
```
and then we can update the save function like this

```kt
private fun saveClicks(): Observable<MedicineFrequencyEvent> {
  return saveMedicineFrequencyButton
      .clicks()
      .map {
        val medicineFrequency = radioIdToMedicineFrequency.getValue(medicineFrequencyRadioGroup.checkedRadioButtonId)
        SaveMedicineFrequencyClicked(medicineFrequency as MedicineFrequencies)
      }
}
```

Reply:
I think the radio button selection will be passed, as we're updating the same variable but this implementation seems fairly better. Wdyt?